### DISCUSSION TITLE
LogIn feature

### DISCUSSION THREAD

**Discussion by Cramses** [webstudio-is/webstudio-community#91] [Wishlist]
A possibility to make user accounts, logins and sites with limited accessibility.

**Comment by Cramses** [comment]
Maybe with AuthO integration or something similar?

**Comment by hidech** [comment]
It would be so nice if we can login with Supabase authentication.

**Reply by ghost** [reply]
For sure it would be great! Hope this feature is coming to Webstudio soon...

**Comment by kof** [comment]
We have 2 templates now in the marketplace that show how to create a login with supabase and firebase. We currently have no plans to become an auth provider ourselves, but we are also happy to add more templates or make it easier in the future.

**Comment by wmlutz** [comment]
I am looking to build a hobby project that requires a login feature. Just came across this post, @kof did marketplace go away? I can't find anything.

**Reply by TrySound** [reply]
Everything is there
<img width="292" alt="image" src="https://github.com/user-attachments/assets/20db63b9-9cad-4c2f-b1d3-7b4471ce8f79">

**Comment by GaneshDholi** [comment]
I think this could be solved with an identity provider like Auth0 (or similar tools such as Keycloak, Firebase Auth, or Okta). These platforms handle user accounts, logins, and session management for you, and also support role-based access control (RBAC) so you can define which users are allowed to access specific parts of your site or APIs. The flow is straightforward: when a user logs in, the IdP issues a JWT token that your backend can verify, and based on the roles/permissions inside that token you can allow or restrict access to certain routes. Auth0 is great if you want a managed, fast setup, while Keycloak is a good self-hosted open-source option, and Firebase/Supabase are lighter developer-friendly choices. This way, you don’t need to reinvent authentication and authorization — instead, you can plug into a proven system and focus on building your app.
